airflow/utils/logging.py

```diff
@@ -60,7 +60,8 @@ class S3Log(object):
     def __init__(self):
         remote_conn_id = configuration.get('core', 'REMOTE_LOG_CONN_ID')
         try:
-            from airflow.hooks.S3_hook import S3Hook
+            from airflow.contrib.hooks.s3_hook import S3Hook
```

Comment:
Agreed about not relying on contrib in core. It will take a while but we can move any dependencies on contrib into abstractions like you and Allison did for logging. This change seems a bit risky to me, I'm not sure if they are 100% compatible (though I'm not blocking it). We might want the abstraction to be created first.